c2rust-transpile/src/lib.rs

```diff
@@ -227,10 +227,65 @@ fn get_module_name(
 
 /// Main entry point to transpiler. Called from CLI tools with the result of
 /// clap::App::get_matches().
-pub fn transpile(tcfg: TranspilerConfig, cc_db: &Path, extra_clang_args: &[&str]) {
+pub fn transpile(tcfg: TranspilerConfig, source_or_cc_db: &Path, extra_clang_args: &[&str]) {
     diagnostics::init(tcfg.enabled_warnings.clone(), tcfg.log_level);
 
-    let lcmds = get_compile_commands(cc_db, &tcfg.filter).unwrap_or_else(|_| {
+    let build_dir = get_build_dir(&tcfg, source_or_cc_db);
+    let mut temp_json_path = std::env::temp_dir();
+    temp_json_path.push("compile_commands.json");
+
+    let ext = source_or_cc_db.extension().unwrap_or_default();
+    let is_c_or_h_file = ext == "c" || ext == "h";
+    let is_wildcard = source_or_cc_db
+        .file_name()
+        .unwrap_or_default()
+        .to_string_lossy()
+        .contains('*');
+
+    let format_compile_command = |build_dir: &Path, file_path: &Path| -> String {
+        format!(
+            r#"{{
+                "directory": "{}",
+                "command": "clang {}",
+                "file": "{}"
+            }}"#,
+            build_dir.to_str().unwrap(),
+            file_path.to_str().unwrap(),
+            file_path.to_str().unwrap()
+        )
+    };
+
+    let mut temp_json = Vec::new();
+    if is_c_or_h_file {
+        let command = format_compile_command(&build_dir, source_or_cc_db);
+        temp_json.push(command);
+    } else if is_wildcard {
+        for entry in read_dir(build_dir.clone()).unwrap() {
+            if let Ok(entry) = entry {
+                let path = entry.path();
+                let ext = path.extension().unwrap_or_default();
+                if ext == "c" || ext == "h" {
+                    let command = format_compile_command(&build_dir, &path);
+                    temp_json.push(command);
+                }
+            }
+        }
```

Comment:
Do we need to handroll globbing?  There's a lot of intricacies with that that we're gonna get correct, so it might be better to just accept multiple sources and let the shell do the glob expansion.

Reply:
I agree. The shell expands wildcards by default so expecting a list of concrete file paths seems simpler.